view/frontend/templates/iframe-handler.phtml

```diff
@@ -0,0 +1,72 @@
+/** @var Script $viewModel */
+$viewModel = $block->getData('view_model');
+
+// Prepare translations
+$acceptText = __('accept %1 cookies', 'marketing');
+$consentMessage = __('Please %1 to view this %2 content.');
+
+$jsTranslations = [
+    'acceptText' => $acceptText,
+    'consentMessage' => $consentMessage,
+];
+?>
+
+<script type="text/javascript" data-cookieconsent="ignore">
+document.addEventListener("DOMContentLoaded", function() {
+    ((document, selector, consentType, translations) => {
+        const createTextNode = text => document.createTextNode(text);
+        const createElement = element => document.createElement(element);
+        
+        document.querySelectorAll(selector).forEach(iframe => {
+            const linkElement = createElement("a");
+            const divElement = createElement("div");
+            const paragraphElement = createElement("p");
+            const iframeSrc = iframe.dataset.cookieblockSrc;
+            
+            // Detect service provider type
+            const serviceProvider = 
+                /google\.com\/maps\/embed/.test(iframeSrc) ? "Google Maps" :
+                /player\.vimeo\.com\/video\//.test(iframeSrc) ? "Vimeo" :
+                /youtube(-nocookie)?\.com\/embed\//.test(iframeSrc) ? "YouTube" : 
+                undefined;
+            
+            if (!serviceProvider) {
+                return;
+            }
+            
+            const iframeHeight = iframe.getBoundingClientRect().height;
+            const iframeWidth = iframe.getBoundingClientRect().width;
+            
+            // Create placeholder content
+            divElement.innerHTML = `
+                <div style="background-color:#CCC;display:inline-block;height:${iframeHeight}px;position:relative;width:${iframeWidth}px;">
+                    <div style="background-color:#848484;border-radius:15px;height:50%;position:absolute;transform:translate(50%,50%);width:50%;">
+                        <p style="color:#FFF;font-size:7.5em;position:relative;top:50%;left:50%;margin:0;text-align:center;transform:translate(-50%,-50%);">&ctdot;</p>
+                    </div>
+                </div>
+            `;
+            
+            // Setup consent message and button
+            divElement.classList.add(`cookieconsent-optout-${consentType}`);
+            linkElement.textContent = translations.acceptText.replace('%1', consentType);
+            linkElement.href = "javascript:Cookiebot.renew()";
+            const message = translations.consentMessage
+                .replace('%1', linkElement.outerHTML)
+                .replace('%2', serviceProvider);
+            paragraphElement.innerHTML = message;
+            
+            divElement.style.fontSize = "1.4rem";
+            divElement.append(paragraphElement);
+            iframe.parentNode.insertBefore(divElement, iframe);
+        });
+    })(document, "iframe[data-cookieblock-src]", "marketing", <?php echo json_encode($jsTranslations); ?>);
```

Comment:
There is no need to JSON encode the texts. You can just output them at the respective spot in the code via PHP.